### PR TITLE
IT Authd/Wazuh DB: Refactor for failures caused by new development of Wazuh DB

### PR DESCRIPTION
**Pull request by Deblintrake09**
# Description
In this PR there were fixes done to tests on the `authd` and `wazuhdb` modules, caused by the development done on Wazuh DB.
<table>
  <tr>
    <th>Related issues</th>
    <th><a href="[url](https://github.com/wazuh/wazuh-qa/issues/2584)">#2584</a></th>
    <th><a href="[url](https://github.com/wazuh/wazuh/issues/10771)">#10771 (Wazuh)</a></th>
  </tr>
</table>

## Work done on `authd` module (test_authd folder):

- fixed messages for `test_authd_worker.py`
- fixed messages for `test_authd.py`
- fixed messages for `test_authd_key_hahs.py`
- Refactored test_authd_agents_ctx

##  Work done on `Wazuh DB` module (test_wazuh_db.py):

- fixed messages for `global_messages` (modify responses and remove unused cases)
- fixed messages for `agent_messages` (modify responses and remove unused cases)
- fixed test_wazuh_db_range_checksum fixture
        
--- 

## Configuration options
|OS | Version | Branch | Type |
|--|--|--|--|
| Centos | 4.4.0 | dev-10771-agent-groups-files-to-wazuh-db | Manager|

**Local Internal Options:** 

> wazuh_modules.debug=2
> monitord.rotate_log=0

---

## Tests
| Type | Jenkins | Local Results | Date | By |Notest|
|--|--|--|--|--|--|
| `test_wazuh_db/`  | [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/24873/) [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/24874/) [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/24875/) | [:green_circle: ](https://github.com/wazuh/wazuh-qa/files/8446133/R1.zip)  [:green_circle: ](https://github.com/wazuh/wazuh-qa/files/8446134/R2.zip)  [:green_circle: ](https://github.com/wazuh/wazuh-qa/files/8446135/R3.zip) | 2022/04/07 | @Deblintrake09  |  
| `test_authd/`  | [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/24818/) [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/24830/) [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/24831/) | [:green_circle: ](https://github.com/wazuh/wazuh-qa/files/8446101/R1.zip) [:green_circle: ](https://github.com/wazuh/wazuh-qa/files/8446104/R2.zip) [:green_circle: ](https://github.com/wazuh/wazuh-qa/files/8446105/R3.zip) | 2022/04/07 | @Deblintrake09  |  
---
## Required
- [x] Proven that tests **pass** when they have to pass.
- [x] Proven that tests **fail** when they have to fail.
- [x] Python codebase satisfies PEP-8 style style guide. `pycodestyle --max-line-length=120 --show-source --show-pep8 file.py`.
- [x] Python codebase is documented following the Google Style for Python docstrings.

### CODE REVIEW

File: tests/integration/test_authd/test_authd_agents_ctx.py [hunk 1]
```diff
@@ -78,14 +81,50 @@
...
+def remove_groups():
+    time.sleep(3)
+    response_token = get_token_login_api(API_PROTOCOL,API_HOST,API_PORT,API_USER,API_PASS,API_LOGIN_ENDPOINT, timeout,login_attempts,sleep)
+    headers = {'Authorization': f"Bearer {response_token}",}
+    params = (
+        ('pretty', 'true'),
+        ('groups_list', 'all'),
+    )
+    response = requests.delete('https://localhost:55000/groups', headers=headers, params=params, verify=False)
+    return response
```
[the remark this answers is not in the document]
Comment (CamiRomero) [reply tests/integration/test_authd/test_authd_agents_ctx.py:127]: Also, please check the function `make_api_call` in `deps/wazuh_testing/wazuh_testing/tools/system.py` as an example. We need one generic function

**Review by CamiRomero — CHANGES_REQUESTED** [review]

File: tests/integration/test_authd/test_authd.py [hunk 2]
```diff
@@ -95,7 +94,7 @@ def get_configuration(request):
 
 
 def test_ossec_auth_messages(get_configuration, set_up_groups, configure_environment, configure_sockets_environment,
-                             clean_client_keys_file_module, restart_authd, wait_for_authd_startup_module,
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply tests/integration/test_authd/test_authd.py:98]: The function removed `restart_authd` has been removed from all authd tests. There is no need to create a new Issue.
The generic function could be applied to other tests, where wazuh or one of its daemons is restarted.

[the remark this answers is not in the document]
Comment (CamiRomero) [reply tests/integration/test_authd/test_authd.py:98]: Yes, you are right, but in this PR only applied the changes to use `restart_wazuh_daemon` and `restart_wazuh_daemon_function` but we need to delete the functions `restart_authd` and `restart_authd_function` located in `wazuh-qa/tests/integration/test_authd/conftest.py` and we need to check in all the repository if we have other functions that can be replace with the generic function created here

**Comment by Deblintrake09** [comment]
### Update 2022/04/12

- Upload only `test_authd` results, that was the modules were changes were requested.

## Tests
| Type  | Local Results | Date | By |
|--|--|--|--|
| `test_authd/`  | [:green_circle: ](https://github.com/wazuh/wazuh-qa/files/8477010/R1.zip) [:green_circle: ](https://github.com/wazuh/wazuh-qa/files/8477011/R2.zip) [:green_circle: ](https://github.com/wazuh/wazuh-qa/files/8477012/R3.zip) | 2022/04/12 | @Deblintrake09  |

File: tests/integration/test_authd/test_authd_agents_ctx.py [hunk 3]
```diff
@@ -78,14 +81,50 @@
...
+
+
+def set_up_groups(groups_list):
+    time.sleep(3)
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply tests/integration/test_authd/test_authd_agents_ctx.py]: fixed

File: tests/integration/test_authd/test_authd_agents_ctx.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply tests/integration/test_authd/test_authd_agents_ctx.py]: removed functions to `deps/wazuh_testing/api.py` file

File: tests/integration/test_authd/test_authd_key_hash.py [hunk 4]
```diff
@@ -168,7 +168,7 @@ def test_ossec_auth_messages_with_key_hash(configure_environment, configure_sock
         while response == '':
             response = receiver_sockets[0].receive().decode()
             if time.time() > timeout:
-                raise ConnectionResetError('Manager did not respond to sent message!')
+                assert response != '', 'Manager did not respond to sent message!'
```
Comment (damarisg) [inline tests/integration/test_authd/test_authd_key_hash.py:171]:
```suggestion
                assert response != '', 'The manager did not respond to the sent message.'
```

File: tests/integration/test_authd/test_authd_valid_name_ip.py [hunk 5]
```diff
@@ -155,7 +153,7 @@ def test_authd_force_options(get_configuration, configure_environment, configure
         while response == '':
             response = receiver_sockets[0].receive().decode()
             if time.time() > timeout:
-                raise ConnectionResetError('Manager did not respond to sent message!')
+                assert response != '', 'Manager did not respond to sent message!'
```
Comment (damarisg) [inline tests/integration/test_authd/test_authd_valid_name_ip.py:156]:
```suggestion
                assert response != '', 'The manager did not respond to the sent message.'
```

**Review by damarisg — CHANGES_REQUESTED** [review]

File: tests/integration/test_authd/test_authd_agents_ctx.py [hunk 6]
```diff
@@ -159,11 +188,12 @@ def clean_agents_timestamp():
...
 
 def check_agent_groups(id, expected, timeout=30):
-    group_path = os.path.join(WAZUH_PATH, 'queue', 'agent-groups', id)
+    subprocess.call(['/var/ossec/bin/agent_groups', '-s', '-i', id, '-q'])
     wait = time.time() + timeout
     while time.time() < wait:
-        ret = os.path.exists(group_path)
-        if ret == expected:
+        groups_created = subprocess.check_output("/var/ossec/bin/agent_groups")
+        print(str(groups_created))
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply tests/integration/test_authd/test_authd_agents_ctx.py:153]: fixed

**Review by damarisg — APPROVED** [review]
## Configuration options
|OS | Version | Branch | Type |
|--|--|--|--|
| Centos | 4.4.0 | dev-10771-agent-groups-files-to-wazuh-db | Manager|


## Tests Results

## Tests
|Test Path | Type | Jenkins - Status  | Date | By |
|--|--|--|--|--|
| `test_wazuh_db/`| Manager | [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/25090/) [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/25095/) [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/25119/) | 2022/04/18| Seyla  |  
| `test_authd/`  |  Manager | [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/25093/) [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/25092/) [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/25120/) | 2022/04/18| Seyla  |

**Pull request merged**